catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -97,7 +97,15 @@ def get_record_data(self, data: dict) -> dict | None:
     @raise_if_empty
     def _get_image_url(self, data: dict) -> str | None:
         group = data.get("edmIsShownBy")
-        return group[0] if group else None
+        if not group:
+            return None
+        url = group[0]
+        # Some Europeana URLs may have prefixes, or reference Dropbox (which we can't
+        # include in our catalog because we cannot access them directly ourselves).
```

Comment:
And this specific Dropbox link is actually broken 😆

Reply:
Does this link actually contain the "L-APC248-" prefix? If so, maybe we need to validate the scheme here? Something like
```py
from urllib.parse import urlparse
result = urlparse(url)
if result.scheme not in ["http", "https"] or not result.netloc:
    return None
```

Not blocking because this doesn't happen often, IIRC.

Reply:
It does, unfortunately 😅 Quoting from the PR description here:

> Because we automatically add the scheme if it isn't present when attempting to validate the URL, I didn't want to add any machinery to check if the scheme was missing from the beginning of the URL (due to the diversity of sources Europeana includes, I can't check all possible URLs we might get returned back).

It's possible that some other links we receive from Europeana are valid but don't include a scheme. Since this was a rare case with a very specific cause, I'm hesitant to change any of the other Europeana machinery without understanding the impact it might have on ingesting all of the sources!